src/manifest.js

```diff
@@ -100,17 +100,27 @@ export const setupMediaPlaylist = ({ playlist, uri, id }) => {
  * @param {Object} master
  *        The master playlist
  */
-export const setupMediaPlaylists = (master) => {
+export const setupMediaPlaylists = (master, type) => {
   let i = master.playlists.length;
 
   while (i--) {
     const playlist = master.playlists[i];
+    const createId = createPlaylistID(i, playlist.uri);
+    let id = createId;
+
+    // DASH Representations can change order across refreshes which can make referring to them by index not work
+    // Instead, use the provided id, available via the NAME attribute.
+    if (type === 'dash') {
+      id = playlist.attributes && playlist.attributes.NAME || id;
+    }
 
     setupMediaPlaylist({
       playlist,
-      id: createPlaylistID(i, playlist.uri)
+      id
     });
     playlist.resolvedUri = resolveUrl(master.uri, playlist.uri);
+    // make sure that if a DASH is used, the old "createId" id is also available
+    master.playlists[createId] = playlist;
```

Comment:
What do we keep this ID around for for DASH?

Reply:
For backwards compatibility.